test/Storage.js

```diff
@@ -0,0 +1,353 @@
+        mockRequest({ endpoint: '/data', query, responseCode: 200, responseBody: skippedData });
+
+        const response = await storage.listData(query);
+
+        expect(response).to.deep.equal(skippedData);
+      });
+    });
+
+    describe('with order and orderBy query', () => {
+      it('should build a valid query string', async () => {
+        const storage = new Storage(settings);
+        const expectedQuery = { order: -1, orderBy: 'timestamp' };
+        mockRequest({ endpoint: '/data', query: expectedQuery, responseCode: 200 });
+
+        await storage.listData(expectedQuery);
+
+        // nock library validates the request by intercepting NodeJS HTTP request and verifying its properties
+      });
+
+      it('should return an array of data objects descending ordered by timestamp', async () => {
+        const storage = new Storage(settings);
+        const data = dataList.reverse();
+        const query = { order: -1, orderBy: 'timestamp' };
+        mockRequest({ endpoint: '/data', query, responseCode: 200, responseBody: data });
+
+        const response = await storage.listData(query);
+
+        expect(response).to.deep.equal(data);
+      });
+    });
+
+    describe('when an error happens', () => {
+      it('should pass error to client', async () => {
+        const storage = new Storage(settings);
+        mockRequest({ endpoint: '/data', responseCode: 500 });
+
+          try {
+            await storage.listData();
+          } catch (err) {
+            expect(err).to.be.an.instanceof(Error);
+          }
+      });
+    });
+  });
+
+  describe('listDataByDevice', () => {
+    const deviceId = '0643ca1462f94b79';
+
+    describe('when deviceId isn\'t provided', () => {
+      it('should throw an exception', async () => {
+        const storage = new Storage(settings);
+        expect(storage.listDataByDevice())
+          .to.eventually.be.rejected;
+      });
+    });
+
+    describe('with empty query', () => {
+      it('should return an array of data objects', async () => {
+        const storage = new Storage(settings);
+        mockRequest({ endpoint: `/data/${deviceId}`, responseCode: 200, responseBody: dataList })
+
+        const response = await storage.listDataByDevice(deviceId);
+
+        expect(response).to.deep.equal(dataList);
+      });
+    });
+
+    describe('with date interval query', () => {
+      it('should build a valid query string', async () => {
+        const storage = new Storage(settings);
+        const expectedQuery = { startDate: '2019-03-18T14:42:03.192Z' };
+        mockRequest({ endpoint: `/data/${deviceId}`, query: expectedQuery, responseCode: 200 });
+
+        await storage.listDataByDevice(deviceId, expectedQuery);
+
+        // nock library validates the request by intercepting NodeJS HTTP request and verifying its properties
+      });
+
+      it('should return an array of data objects between time interval', async () => {
```

Comment:
In this test you are actually validating that the library returns what is returned by the server when it succeeds. You could rename it to make it explicit and remove the others that are doing the same (e.g. "should return an array of two data objects skipped by 3", ...)

Reply:
Done. I've kept only the first test doing this validation (returns what is returned by the server).